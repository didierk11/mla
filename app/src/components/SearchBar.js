const React = require("react");

class SearchBar extends React.Component {
  constructor() {
    super();
    this.state = { searchTerm: "" };
  }

  handleSearchTerm(event) {
    this.setState({ searchTerm: event.target.value });
  }

  render() {
    return (
      <div className="navi">
        <div className="container-fluid">
          <div className="row justify-content-sm-center">
            <div className="col-sm-11 ">
              <div className="navi-logo"></div>
              <form action="/items" method="get">
                <div className="navi-bar">
                  <div class="navi-search">
                    <input
                      type="text"
                      onChange={this.handleSearchTerm.bind(this)}
                      placeholder="Nunca dejes de buscar"
                    />
                    <input
                      type="hidden"
                      value={this.state.searchTerm}
                      name="search"
                    />
                  </div>
                  <div class="search-submit">
                    <button type="submit"></button>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default SearchBar;
