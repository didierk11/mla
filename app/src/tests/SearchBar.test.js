import React from "react";
import { render } from "react-dom";
import SearchBar from "../components/SearchBar";
import renderer from "react-test-renderer";

describe("Snapshot test:", () => {
  it("SearchBar render correctly", () => {
    const div = renderer.create(<SearchBar />).toJSON();
    expect(div).toMatchSnapshot();
  });
});

describe("Smoke test:", () => {
  // SearchBar
  it("SearchBar render without crashing", () => {
    const div = document.createElement("div");
    render(<SearchBar />, div);
  });
});
