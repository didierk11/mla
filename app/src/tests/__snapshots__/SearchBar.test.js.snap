// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test: SearchBar render correctly 1`] = `
<div
  className="navi"
>
  <div
    className="container-fluid"
  >
    <div
      className="row justify-content-sm-center"
    >
      <div
        className="col-sm-10 "
      >
        <div
          className="navi-logo"
        />
        <form
          action="/items"
          method="get"
        >
          <div
            className="navi-bar"
          >
            <div
              className="navi-search"
            >
              <input
                onChange={[Function]}
                placeholder="Nunca dejes de buscar"
                type="text"
              />
              <input
                name="search"
                type="hidden"
                value=""
              />
            </div>
            <div
              className="search-submit"
            >
              <button
                type="submit"
              />
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
